sdk/core/azure-core/test/ut/transport_adapter.cpp

```diff
@@ -30,461 +30,454 @@ namespace Azure { namespace Core { namespace Test {
   Azure::Core::Http::HttpPipeline TransportAdapter::pipeline(policies);
   Azure::Core::Context TransportAdapter::context = Azure::Core::GetApplicationContext();
 
-  // multiThread test requires `ConnectionsOnPool` hook which is only available when building
-  // TESTING_BUILD. This test is only built when that case is true.
-  TEST_F(TransportAdapter, getMultiThread)
-  {
-    Azure::Core::Http::Url host("http://httpbin.org/get");
-    Azure::Core::Http::CurlConnectionPool::ClearIndex();
+// Connection pool feature is curl-implementation only. No other transport adapter would have the
+// connection pool
+#ifdef BUILD_CURL_HTTP_TRANSPORT_ADAPTER
+    // multiThread test requires `ConnectionsOnPool` hook which is only available when building
+    // TESTING_BUILD. This test is only built when that case is true.
+    TEST_F(TransportAdapter, connectionPoolTest)
+    {
+      Azure::Core::Http::Url host("http://httpbin.org/get");
+      Azure::Core::Http::CurlConnectionPool::ClearIndex();
+
+      auto threadRoutine = [host]() {
+        auto request = Azure::Core::Http::Request(Azure::Core::Http::HttpMethod::Get, host);
+        auto response = pipeline.Send(context, request);
+        checkResponseCode(response->GetStatusCode());
+        auto expectedResponseBodySize = std::stoull(response->GetHeaders().at("content-length"));
+        CheckBodyFromBuffer(*response, expectedResponseBodySize);
+      };
+
+      std::thread t1(threadRoutine);
+      std::thread t2(threadRoutine);
+      t1.join();
+      t2.join();
+
+      // 2 connections must be available at this point
+      EXPECT_EQ(Http::CurlConnectionPool::ConnectionsOnPool("httpbin.org"), 2);
+
+      std::thread t3(threadRoutine);
+      std::thread t4(threadRoutine);
+      std::thread t5(threadRoutine);
+      t3.join();
+      t4.join();
+      t5.join();
+
+      // Two connections re-used plus one connection created
+      EXPECT_EQ(Http::CurlConnectionPool::ConnectionsOnPool("httpbin.org"), 3);
+
+#ifdef RUN_LONG_UNIT_TESTS
+      {
+        // Test pool clean routine
+        std::cout << "Running Connection Pool Cleaner Test. This test takes more than 3 minutes to "
+                     "complete."
+                  << std::endl
+                  << "Add compiler option -DRUN_LONG_UNIT_TESTS=OFF when building if you want to "
+                     "skip this "
+                     "test."
+                  << std::endl;
+
+        // Wait for 180 secs to make sure any previous connection is removed by the cleaner
+        std::this_thread::sleep_for(std::chrono::milliseconds(1000 * 180));
+
+        std::cout << "First wait time done. Validating state." << std::endl;
+
+        // index is not affected by cleaner. It does not remove index
+        EXPECT_EQ(Http::CurlConnectionPool::ConnectionsIndexOnPool(), 1);
```

Comment:
What's index and when is it set to 1?

Reply:
index is the table of host names inside the pool.
The connection pool internally creates one connection pool for each host name.
For instance, it starts with index 0 (no connections)
Then someone asks for a new connection to abc.com , that adds an index "abc.com" to the pool and a new connection to that index.
If another connection is asked to "abc.com" and that index contains 0 connnections (previous connection haven't been returned), a new conection is added to the index
Later, if both connections came back to the pool, the index will now contain 2 connections.  
So, one index, and 2 connections.
If someone else ask a connection to xyz.com, a new index will be created.

That's how connection pool works.

The clean routine does not affect the index. When it find an index with 0 connections, it does not remove because there might be a connection that will come back and we would need to insert the index again.